pymodm/fields.py

```diff
@@ -1180,15 +1199,23 @@ def contribute_to_class(self, cls, name):
                                                     self._on_delete)
 
     def to_python(self, value):
+        doc = value
         if isinstance(value, dict):
             # Try to convert the value into our document type.
-            return self.related_model.from_document(value)
-        elif isinstance(value, self.related_model):
-            return value
+            try:
+                doc = self.related_model.from_document(value)
```

Comment:
Well we can but that requires to change ``from_document``. And maybe that is the right way.
Here https://github.com/mongodb/pymodm/blob/ce603f5079fb602eeb2421e1bca767f4cb2bd8ac/pymodm/base/models.py#L265 if there is a ``_cls`` field in value we create a ``cls`` from that field and it does not check if the ``cls`` is subclass of current ``cls`` so we can do something like this without any exception:
```python
post = User.from_document({'title': 'I am a Post', '_cls': 'Post'})
assert isinstance(post, Post) == True
```
Without these changes we should check if ``doc`` is instance of ``related_model`` or not. It could be a reference id in form of Document.